vmdk_plugin/utils/plugin_utils/plugin_utils.go

```diff
@@ -0,0 +1,147 @@
+// limitations under the License.
+
+package plugin_utils
+
+// This file holds utility/helper methods required in plugin module
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/vmware/docker-volume-vsphere/vmdk_plugin/drivers"
+)
+
+const (
+	// consts for finding and parsing linux mount information
+	linuxMountsFile = "/proc/mounts"
+
+	// index datastore from volume meta
+	// "datastore" key is defined in vmdkops service
+	datastoreKey = "datastore"
+)
+
+// VolumeInfo - Volume fullname, datastore and metadata
+type VolumeInfo struct {
+	VolumeName    string
+	DatastoreName string
+	VolumeMeta    map[string]interface{}
+}
+
+// GetMountInfo - return a map of mounted volumes and devices
+func GetMountInfo(mountRoot string) (map[string]string, error) {
+	volumeMountMap := make(map[string]string) //map [volume mount path] -> device
+	data, err := ioutil.ReadFile(linuxMountsFile)
+
+	if err != nil {
+		log.Errorf("Can't get info from %s (%v)", linuxMountsFile, err)
+		return volumeMountMap, err
+	}
+
+	for _, line := range strings.Split(string(data), "\n") {
+		field := strings.Fields(line)
+		if len(field) < 2 {
+			continue // skip empty line and lines too short to have our mount
+		}
+		// fields format: [/dev/sdb /mnt/vmdk/vol1 ext2 rw,relatime 0 0]
+		if filepath.Dir(field[1]) != mountRoot {
+			continue
+		}
+		volumeMountMap[filepath.Base(field[1])] = field[0]
+	}
+	return volumeMountMap, nil
+}
+
+// AlreadyMounted - check if volume is already mounted on the mountRoot
+func AlreadyMounted(name string, mountRoot string) bool {
+	volumeMap, err := GetMountInfo(mountRoot)
+
+	if err != nil {
+		return false
+	}
+
+	if _, ok := volumeMap[name]; ok {
+		return true
+	}
+	return false
+}
+
+// JoinVolName - return a full name in format volume@datastore
+func JoinVolName(volName string, datastoreName string) string {
+	return strings.Join([]string{volName, datastoreName}, "@")
+}
+
+// SplitVolName - split a volume name into short name and datastore
+func SplitVolName(fullVolName string) []string {
+	return strings.Split(fullVolName, "@")
+}
+
+// IsFullVolName - Check if volume name is full volume name
+func IsFullVolName(volName string) bool {
+	return strings.ContainsAny(volName, "@")
+}
+
+// GetNameFromRefmap - traverse the name entries in refmap
+// and if a single entry with volName match is found(no collision),
+// use that name as full volume name
+func GetNameFromRefmap(volName string, d drivers.VolumeDriver) string {
+	volumeNameList := d.VolumesInRefMap()
+
+	count := 0
+	fullname := ""
+
+	for _, name := range volumeNameList {
+		refVolName := SplitVolName(name)[0]
+		if refVolName != volName {
+			continue
+		}
+		// if there are collisions, return
+		if count > 0 {
+			return ""
+		}
+		count++
+		fullname = name
+	}
+	return fullname
+}
+
+// GetVolumeInfo - return VolumeInfo which a qualified volume name,
+// datastore name volume metadata if retrieved
```

Comment:
How about changing comment to following?

return VolumeInfo with a qualified volume name.
Optionally returns datastore and volume metadata if retrieved from ESX.
If Volume Metadata is nil then caller can use getVolume()